src/Illuminate/Database/Connectors/ConnectionFactory.php

```diff
@@ -96,6 +97,55 @@ protected function createReadPdo(array $config)
      */
     protected function createPdoResolver(array $config)
     {
+        if (array_key_exists('host', $config)) {
+            return $this->createPdoResolverWithHosts($config);
+        }
+
+        return $this->createPdoResolverWithoutHosts($config);
+    }
+
+    /**
+     * Create a new Closure that resolves to a PDO instance with a specific host or an array of hosts.
+     *
+     * @param  array  $config
+     * @return \Closure
+     */
+    protected function createPdoResolverWithHosts(array $config)
+    {
+        return function () use ($config) {
+            if (! is_array($config['host'])) {
+                $hosts = [$config['host']];
+            } else {
+                $hosts = $config['host'];
+                shuffle($hosts);
+            }
+
+            foreach ($hosts as $host) {
+                $config['host'] = $host;
+
+                try {
+                    return $this->createConnector($config)->connect($config);
+                } catch (PDOException $e) {
```

Comment:
Would this suppress any other exceptions? Such as the user not having access to the database or the database not existing? If so I'd think we would want to be much more narrow in scope and report all but the actual connection error.